Rename SSL[_CTX]_add1_CA_list -> SSL[_CTX]_add1_to_CA_list

levitte opened the pull request:
They add a single item, so the names give a false impression of what
they do, making them hard to remember.  Better to give them a somewhat
better name.

Fixes #6930

levitte committed:
Rename SSL[_CTX]_add1_CA_list -> SSL[_CTX]_add1_CA

They add a single item, so the names give a false impression of what
they do, making them hard to remember.  Better to give them a somewhat
better name.

Fixes #6930

levitte commented:
These names are new in upcoming 1.1.1, so a rename won't break anything.

daurnimator reviewed (approved):
Looks good!

levitte commented:
That's a separate issue, and considering we're *very* late in the 1.1.1 cycle and in a general feature freeze, it's not likely to go through for it.  *But*, PRs still welcome, to be merged post-1.1.1.  Feel free to submit!

daurnimator commented:
> That's a separate issue, and considering we're very late in the 1.1.1 cycle and in a general feature freeze, it's not likely to go through for it. But, PRs still welcome, to be merged post-1.1.1. Feel free to submit!

I have a couple of other issues I'd like to rush through (#6709 and #2673): this item can be at least done with a simple compat-wrapper, so I don't mind if it doesn't make it into this release.

t-j-h commented:
It is not adding-a-list - it is adding-to-a-list - and it is consistently named with all the other functions that deal with the CA_list.

I don't think this should go in.

And SSL_add_client_CA and SSL_CTX_add_client_CA already exist as functions that do the same thing.
The set0, get0 and add1 versions were added in commit fa7c263747cb73f03b321399a1452cc40516d9a4

levitte commented:
The commit message mentions `SSL[_CTX]_add_client_CA` as "The older client_CA functions mainly just call the new versions now"...  that seems to indicate that they should be deprecated at some point.

The rest of the `_CA_list` functions *do* manipulate a whole list (implemented as a stack), so I'll still argue that the names of these two are confusing.

levitte commented:
@t-j-h, I have zero problems calling these two `_add1_to_CA_list`. If everyone's OK with that, I will change the PR accordingly.

t-j-h commented:
I would +1 that.

levitte committed:
squash! Rename SSL[_CTX]_add1_CA_list -> SSL[_CTX]_add1_CA

New commit message should be:

Rename SSL[_CTX]_add1_CA_list -> SSL[_CTX]_add1_to_CA_list

They add a single item, so the names give a false impression of what
they do, making them hard to remember.  Better to give them a somewhat
better name.

Fixes #6930

levitte commented:
Ok fixed.

mattcaswell commented:
Ping @levitte - can this be merged?

levitte commented:
Merged.

64a48fc7f0bc2d3ff587791b93a357bd98e1a5b8 Rename SSL[_CTX]_add1_CA_list -> SSL[_CTX]_add1_to_CA_list
